### PR TITLE
Add test coverage via Codecov

### PR DESCRIPTION
**Pull request by pmacik**
This PR adds a code coverage reporting into tests.

At the end of test output it will print reports to the stdout similar to the following:

```
go tool cover -func=./out/coverage.test-unit.txt
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/common.go:9:                  RequeueOnNotFound       0.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/common.go:17:                 Requeue                 0.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/common.go:22:                 Done                    100.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/controller.go:16:             Add                     0.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/controller.go:21:             newReconciler           0.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/controller.go:26:             add                     0.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/planner/planner.go:40:        searchCRDDescription    77.3%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/planner/planner.go:87:        searchCR                88.9%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/planner/planner.go:111:       Plan                    71.4%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/planner/planner.go:128:       NewPlanner              100.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/reconciler.go:28:             appendEnvFrom           50.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/reconciler.go:55:             Reconcile               55.9%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/retriever.go:33:              getNestedValue          90.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/retriever.go:50:              getCRKey                87.5%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/retriever.go:67:              read                    88.2%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/retriever.go:106:             extractSecretItemName   100.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/retriever.go:112:             readSecret              90.9%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/retriever.go:134:             store                   100.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/retriever.go:143:             saveDataOnSecret        80.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/retriever.go:165:             Retrieve                80.0%
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/retriever.go:186:             NewRetriever            100.0%
total:                                                                                                                  (statements)            67.9%
```

The code coverage data will be used later on to upload to [Codevov](https://codecov.io).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -295,3 +309,40 @@ deploy: deploy-rbac deploy-crds
...
+	@echo "uploading test coverage report after PR was merged..."
+	bash <(curl -s https://codecov.io/bash) \
+		-t $(CODECOV_TOKEN) \
+		-f $(GOCOV_DIR)/*.txt \
+		-C $(BASE_COMMIT) \
+		-r $(REPO_OWNER)/$(REPO_NAME) \
+		-Z
+endif
+
+CODECOV_TOKEN := "037da9d0-5a3c-49cb-9774-f2281af621fb"
```
Comment (sbose78) [inline Makefile]: Hope this isn't real. How are you replacing this in our CI , @pmacik ?

Comment (pmacik) [reply Makefile]: Replacing the token with the real one is still something that I'm still working. I need to discuss it with the openshift-ci/prow team.

**Comment by pmacik** [comment]
Token synced to the CI: https://github.com/openshift/release/pull/4539

File: Makefile [hunk 2]
```diff
@@ -90,6 +90,14 @@ CGO_ENABLED ?= 0
 GO111MODULE ?= on
 GOCACHE ?= "$(shell echo ${PWD})/out/gocache"
 
+# This variable is for artifacts to be archived by Prow jobs at OpenShift CI
+# The actual value will be set by the OpenShift CI accordingly
+ARTIFACT_DIR ?= "$(shell echo ${PWD})/out"
+
+GOCOV_DIR ?= $(ARTIFACT_DIR)/test-coverage
```
[the remark this answers is not in the document]
Comment (pmacik) [reply Makefile]: The directory IS `./out` by default, not the root.

File: Makefile (hunk 1 above)
Comment (pmacik) [reply Makefile]: @sbose78  The token in CI is passed to the test container as a secret/volumemount: https://github.com/openshift/release/pull/4556

**Pull request merged**